src/console.js

```diff
@@ -0,0 +1,69 @@
+/*
+ * This file is part of Privacy Badger <https://www.eff.org/privacybadger>
+ * Copyright (C) 2015 Electronic Frontier Foundation
+ *
+ * Privacy Badger is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * Privacy Badger is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Privacy Badger.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+function logMessage () {
+  var messages = [];
+  var _log = debounce(function () {
+    setTimeout(function () {
+      messages.forEach(function (message) {
+        console.log(message);
+      });
+      messages = [];
+    }, 0);
+  }, 1000);
+  return function (message) {
+    if (message.type === 'LOG' && messages.indexOf(message.text) === -1) {
+      messages.push(message.text);
+      _log();
+    }
+  };
+}
+
+chrome.runtime.onMessage.addListener(logMessage());
+
+// from Underscore v1.6.0
```

Comment:
I'll begin moving the setTimeout and debounce to the background script immediately.

I believe the debounce is necessary because many of the replaced resources are very persistent about retrying to load the resource, so the console message is sent many times and fills up the log.  This is also the reason I've set the debounce interval to a full second and I check for duplicates before logging.

How about `logToTab` instead of `consoleBroadcast`